Remove `lineinfo` from RelWithDebInfo DALI builds

JanuszL opened the pull request:
- as CUDA doesn't provide a tool to strip debug info
  to avoid bloating release binary with the debug info
  that cannot be removed before the release remove
  `lineinfo` from RelWithDebInfo build configuration

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

<!---
Thank you for contributing to NVIDIA DALI! If you haven't yet,
please read the contributing guidelines in the CONTRIBUTING.md file.

We need a few more information from you to proceed.
Please fill the relevant sections in this PR template.

Fields in the Checklist section can be marked after you create and save the Pull Request.
--->


## Category:
<!--- Please pick one from below: --->
<!--- **Bug fix** (*non-breaking change which fixes an issue*) --->
<!--- **New feature** (*non-breaking change which adds functionality*) --->
<!--- **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*) --->
<!--- **Refactoring** (*Redesign of existing code that doesn't affect functionality*) --->
**Other** (*e.g. Documentation, Tests, Configuration*)



## Description:
- as CUDA doesn't provide a tool to strip debug info
  to avoid bloating release binary with the debug info
  that cannot be removed before the release remove
  `lineinfo` from RelWithDebInfo build configuration
<!---
Please explain what kind of change is in this PR and why it is submitted.
Any additional context or description of the solution is welcomed.
Examples:
- It fixes a bug *bug description*
- It adds new feature needed because of *why we need this feature*
- Refactoring to improve *what*
- The *new feature/bugfix* uses *a new data structure/algorithm* to do *X* instead of *Y*.
--->



## Additional information:

### Affected modules and functionalities:
- CMake
<!--- Describe here what was changed, added, removed. --->



### Key points relevant for the review:
- NA
<!--- Describe here what is the most important part that reviewers should focus on. --->



<!--- 
At this point you can hit "Create".
The checklist below shall be filled in the created PR.
--->

## Checklist

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

### DALI team only

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A
<!---  Introduce new or affected requirement IDs, if applicable --->

**JIRA TASK**: DALI-2601
<!--- DALI-XXXX or NA --->

dali-automaton commented:
**CI MESSAGE**: [3969635]: BUILD STARTED

dali-automaton commented:
**CI MESSAGE**: [3969635]: BUILD PASSED

JanuszL commented:
It shrinks CUDA 11.x build from 700MB->400MB.

klecki commented:
Can we split this into configurations that are suitable for building release wheels (without line info) and second one for the developer that want to keep the line info for the purpose of local debugging? Otherwise we would need to manually add the flag every time.

JanuszL commented:
> Can we split this into configurations that are suitable for building release wheels (without line info) and second one for the developer that want to keep the line info for the purpose of local debugging? Otherwise we would need to manually add the flag every time.

I have added an optional flag `RELWITHDEBINFO_CUDA_DEBUG` set to `OFF` by default. I don't think we use this limited debug info for the device code that often anyway. I think in most cases the developers use `Debug` build configuration which uses `-lineinfo`.

dali-automaton commented:
**CI MESSAGE**: [3976486]: BUILD FAILED

dali-automaton commented:
**CI MESSAGE**: [3976486]: BUILD PASSED
